api/src/main/java/org/openmrs/module/aijar/api/impl/AijarServiceImpl.java

```diff
@@ -450,13 +450,17 @@ public boolean isTransferredIn(Patient patient, Date date) {
     public List<Encounter> getTransferHistory(Patient patient) {
 
         EncounterService encounterService = Context.getEncounterService();
+        Collection<EncounterType> encounterTypes = new ArrayList<>();
+        List<Encounter> encounters = new ArrayList<>();
 
-        Collection<EncounterType> encounterTypes = encounterService.findEncounterTypes(TRANSFER_IN.name());
-        encounterTypes.addAll(encounterService.findEncounterTypes(TRANSFER_OUT.name()));
+        encounterTypes.add(encounterService.getEncounterTypeByUuid(TRANSFER_IN.uuid()));
+        encounterTypes.add(encounterService.getEncounterTypeByUuid(TRANSFER_OUT.uuid()));
 
         EncounterSearchCriteria encounterSearchCriteria = new EncounterSearchCriteriaBuilder().setPatient(patient).setIncludeVoided(false).setEncounterTypes(encounterTypes).createEncounterSearchCriteria();
 
-        List<Encounter> encounters = encounterService.getEncounters(encounterSearchCriteria);
+        if (!encounterTypes.isEmpty()) {
```

Comment:
@slubwama This check is never needed since you initialize the encounterTypes above

Reply:
@ssmusoke this is  only reason why we were able to see that logical bug. Just in case the metadata for transfer in/out encounter types is not deployed, the criteria will pick any encounter. Thats why the  check is really necessary

Reply:
Changed it to check in the criteria